com.unity.multiplayer.mlapi/Runtime/Core/NetworkedBehaviour.cs

```diff
@@ -14,19 +14,105 @@
 using MLAPI.Profiling;
 using MLAPI.Reflection;
 using MLAPI.Security;
+using MLAPI.Serialization;
 using MLAPI.Serialization.Pooled;
 using MLAPI.Spawning;
 using BitStream = MLAPI.Serialization.BitStream;
-using MLAPI.Serialization;
 using Unity.Profiling;
 
+#if UNITY_EDITOR
+using System.Runtime.CompilerServices;
+[assembly: InternalsVisibleTo("Unity.Multiplayer.MLAPI.Editor.CodeGen")]
+#endif // UNITY_EDITOR
+
 namespace MLAPI
 {
     /// <summary>
     /// The base class to override to write networked code. Inherits MonoBehaviour
     /// </summary>
     public abstract partial class NetworkedBehaviour : MonoBehaviour
     {
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal enum NExec
+        {
+            None = 0,
+            Server = 1,
+            Client = 2
+        }
+
+        const String StandardRPC_Channel = "STDRPC";
+
+#pragma warning disable 414
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal NExec __nexec = NExec.None;
+#pragma warning restore 414
+
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal BitWriter BeginSendServerRpc(ServerRpcSendParams sendParams, bool isReliable)
+        {
+            // @mfatihmar (Unity) Begin: Temporary, placeholder implementation
+            RPCQueueManager rpcQueueMananger = NetworkingManager.Singleton.GetRPCQueueManager();
+            if(rpcQueueMananger != null)
+            {
+                var writer = rpcQueueMananger.BeginAddQueueItemToOutboundFrame(RPCQueueManager.QueueItemType.ServerRPC, Time.realtimeSinceStartup, StandardRPC_Channel,0, NetworkingManager.Singleton.ServerClientId,null);
+
+                writer.WriteBit(false); // Encrypted
+                writer.WriteBit(false); // Authenticated
+                writer.WriteBits(MLAPIConstants.MLAPI_STD_SERVER_RPC, 6); // MessageType
+                writer.WriteUInt64Packed(NetworkId); // NetworkObjectId
+                writer.WriteUInt16Packed(GetBehaviourId()); // NetworkBehaviourId
+
+                return writer;
+            }
+            return null;
+            // @mfatihmar (Unity) End: Temporary, placeholder implementation
+        }
+
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal void EndSendServerRpc(BitWriter writer, ServerRpcSendParams sendParams, bool isReliable)
+        {
+
+            if (writer == null) return;
```

Comment:
Is this an error condition? Like would crashing here actually be the right thing?

Reply:
Good point.  I believe it has much to do with the previous comment about the BeginServerRPC returning null.
Looking at some of the code generated:
[ServerRpc]
private void PingMySelfServerRPC(int pingnumber)
{
	NetworkingManager singleton = NetworkingManager.Singleton;
	if ((object)singleton == null || !singleton.IsListening)
	{
		return;
	}
	if (__nexec != NExec.Server && (singleton.IsClient || singleton.IsHost))
	{
		ServerRpcSendParams val = default(ServerRpcSendParams);
		BitWriter bitWriter = base.BeginSendServerRpc(val, true);
		if (bitWriter != null)
		{
			bitWriter.WriteUInt32Packed(3021433314u);
			bitWriter.WriteInt32Packed(pingnumber);
		}
		base.EndSendServerRpc(bitWriter, val, true);
	}
	if (__nexec == NExec.Server && (singleton.IsServer || singleton.IsHost))
	{
		Debug.Log("Pinging MySelf:" + pingnumber);
		Counter++;
	}
}

It looks like it checks the writer and if the writer is null then it just returns a null writer to the EndSendServerRpc method which does nothing but returns if the writer is null.  

We might provide a form of warning about this condition if in develop or editor mode.

Reply:
I would say if `bitWriter == null` is never a valid runtime condition, then I'd recommend having the process hard error out

Reply:
I think `bitWriter == null` is an absent state which means there is no writer to be written to and that could happen for different reasons (filtering, asking for it in a weird time/context etc.). why would you think `null` isn't a form of "absent" state? we could do optionals/nullables but null gives us that without any struct containers/wrappings.